relay-server/src/services/project.rs

```diff
@@ -579,52 +579,72 @@ impl Project {
         }
     }
 
-    /// Remove metric buckets that are not allowed to be ingested.
+    /// Remove metric buckets or tags that are not allowed to be ingested.
     fn filter_metrics(&self, metrics: &mut Vec<Bucket>) {
         let Some(state) = &self.state_value() else {
             return;
         };
 
-        Self::apply_metrics_deny_list(&state.config.metrics, metrics);
-        Self::filter_disabled_namespace(state, metrics);
-    }
-
-    fn apply_metrics_deny_list(deny_list: &ErrorBoundary<Metrics>, buckets: &mut Vec<Bucket>) {
-        let ErrorBoundary::Ok(metrics) = deny_list else {
-            return;
-        };
+        let metrics_config = &state.config.metrics;
 
-        buckets.retain(|bucket| {
-            if metrics.denied_names.is_match(&bucket.name) {
+        metrics.retain_mut(|bucket| {
+            if Self::metric_name_denied(metrics_config, bucket) {
```

Comment:
All of these functions are not associated with the `Project` struct in any way. Can we make them free functions?